camayoc/tests/qpc/cli/ocp.py

```diff
@@ -0,0 +1,148 @@
+from camayoc.tests.qpc.cli.utils import scan_job
+from camayoc.tests.qpc.cli.utils import scan_show_and_check
+from camayoc.tests.qpc.cli.utils import scan_start
+from camayoc.tests.qpc.cli.utils import source_show
+from camayoc.tests.qpc.cli.utils import source_show_and_check
+from camayoc.tests.qpc.cli.utils import wait_for_scan
+from camayoc.utils import client_cmd
+
+
+ocp_config = get_ocp_config_info()
+
+
+def check_if_discovery_was_deployed(list_projects):
+    """Check if discovery can be found in deployed projects."""
+    discovery_found = False
+    for project in list_projects:
+        if project.get("name", "") == "discovery":
+            discovery_found = True
+            break
+    return discovery_found
+
+
+def test_end_to_end_cli_ocp(qpc_server_config):
+    """End-to-end test for ocp instance with known deployed project using cli route."""
+    # ocp cred tests
+    # adding cred
+    cred_name = "ocp_cred"
+    cred_add_and_check(
+        {"name": cred_name, "token": None, "type": "openshift"},
+        [(TOKEN_INPUT, utils.uuid4())],
+    )
+    # editing cred
+    qpc_cred_edit = pexpect.spawn("{} cred edit --name={} --token".format(client_cmd, cred_name))
+    assert qpc_cred_edit.expect(TOKEN_INPUT) == 0
+    qpc_cred_edit.sendline(ocp_config["token"])
+    assert qpc_cred_edit.expect('Credential "{}" was updated'.format(cred_name)) == 0
+    assert qpc_cred_edit.expect(pexpect.EOF) == 0
+    qpc_cred_edit.close()
+    assert qpc_cred_edit.exitstatus == 0
+
+    # checking for saved cred
+    cred_show_and_check(
+        {"name": cred_name},
+        show_cred_output(
+            {"auth_token": MASKED_TOKEN_OUTPUT, "cred_type": "openshift", "name": cred_name}
+        ),
+    )
+
+    # ocp source tests
+    # adding source
+    source_name = "ocp_source"
+    hosts = ocp_config["hosts"]
+
+    qpc_source_add = pexpect.spawn(
+        "{} source add --name {} --cred {} --hosts {} --ssl-cert-verify false --type openshift".format(
+            client_cmd, source_name, cred_name, "127.0.0.1"
+        )
+    )
+    assert qpc_source_add.expect('Source "{}" was added'.format(source_name)) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+
+    # editing source
+    qpc_source_edit = pexpect.spawn(
+        "{} source edit --name {} --hosts {}".format(client_cmd, source_name, hosts)
+    )
+    assert qpc_source_edit.expect('Source "{}" was updated'.format(source_name)) == 0
+    assert qpc_source_edit.expect(pexpect.EOF) == 0
+    qpc_source_edit.close()
+    assert qpc_source_edit.exitstatus == 0
+
+    # checking for saved source
+    source_show_and_check(
+        {"name": source_name},
+        show_source_output(
+            {
+                "cred_name": cred_name,
+                "hosts": hosts,
+                "name": source_name,
+                "options": {"ssl_cert_verify": "false"},
```

Comment:
same observation here. ssl_cert_verify should come from camayoc test configuration